xmtp_mls/src/storage/encrypted_store/group.rs

```diff
@@ -82,6 +99,14 @@ impl DbConnection<'_> {
         Ok(self.raw_query(|conn| query.load(conn))?)
     }
 
+    /// Return only the [`Purpose::Sync`] groups
+    pub fn find_sync_groups(&self) -> Result<Vec<StoredGroup>, StorageError> {
```

Comment:
I guess there isn't really a reason to ever look for all sync and conversation groups at the same time, right?

Reply:
right, AFAICT.  I couldn't think of a scnerio where we'd want to. 
And we explicitly want to keep `listGroups()` as it is, for only returning "normal (conversation)" groups.

If we should find a need later to look for all groups at the same time, we'll add it if/when that need arises.